astroplan/constraints.py

```diff
@@ -394,7 +395,81 @@ def compute_constraint(self, times, observer, targets):
                              "MoonSeparationConstraint.")
         return mask
 
+    
+class DarkGreyBrightConstraint(Constraint):
+    """
+    Constrain whether it is dark, gray or bright time or how bright.
+    """
+    # TODO: make a new test
+    @u.quantity_input(horizon=u.deg)
+    def __init__(self, lunar_horizon=90*u.deg, min_moon_illumination=None,
+                                                max_moon_illumination=None):
+        '''
+        Parameters
+        ----------
+        Lunar_horizon : `~astropy.units.Quantity` (optional)
+            The altitude of the ``horizon`` and defaults to 90 degrees       
+        min_moon_illumination : float or `None` (optional)
+            the minimum value of the fraction of the moon illuminated
+        max_moon_illumination : float or `None` (optional)
+            the maximum value of the fraction of the moon illuminated
+        
+        Example
+        -------
+        Constrain the observations to times where the moon :
+
+        >>> from astroplan import Observer
+        >>> from astroplan.constraints import DarkGreyBrightConstraint
+        >>> subaru = Observer.at_site("Subaru")
+        >>> darktime = DarkGreyBrightConstraint(lunar_horizon=-5*u.deg) 
```

Comment:
Thanks Brett, 

This is good to know. I guess that the significant differences in definitions make the classmethods not very useful. 

It does makes the enhancement that @kvyh is working on all the more useful though!